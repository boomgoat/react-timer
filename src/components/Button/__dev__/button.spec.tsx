import React from "react";
import { screen, render } from "@testing-library/react";
import { Button } from "..";

describe("button tests", () => {
  afterAll(() => jest.clearAllMocks());

  test("renders with correct label", () => {
    render(
      <Button
        clickEventHandler={() => {}}
        label="Test"
        dataTestId="test-button"
      />
    );
    const button = screen.getByTestId("test-button");
    expect(button.textContent).toBe("Test");
  });

  test("snapshot test", () => {
    const { asFragment } = render(
      <Button clickEventHandler={() => {}} label="Test" />
    );
    expect(asFragment).toMatchSnapshot();
  });
});
