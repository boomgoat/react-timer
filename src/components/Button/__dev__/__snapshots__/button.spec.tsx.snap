// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button tests snapshot test 1`] = `[Function]`;
